cmd/kube-proxy/app/server_test.go

```diff
@@ -270,73 +269,66 @@ func Test_getProxyMode(t *testing.T) {
 // Config and iptinterface are not nil when CleanupAndExit is true.
 // To avoid proxy crash: https://github.com/kubernetes/kubernetes/pull/14736
 func TestProxyServerWithCleanupAndExit(t *testing.T) {
-	// creates default config
-	config := options.NewProxyConfig()
+	options := KubeProxyOptions{CleanupAndExit: true}
+	config := componentconfig.KubeProxyConfiguration{BindAddress: "0.0.0.0"}
 
-	// sets CleanupAndExit manually
-	config.CleanupAndExit = true
+	proxyserver, err := options.NewProxyServer(&config)
 
-	// creates new proxy server
-	proxyserver, err := NewProxyServerDefault(config)
-
-	// verifies that nothing is nill except error
 	assert.Nil(t, err)
 	assert.NotNil(t, proxyserver)
-	assert.NotNil(t, proxyserver.Config)
 	assert.NotNil(t, proxyserver.IptInterface)
 }
 
 func TestGetConntrackMax(t *testing.T) {
 	ncores := runtime.NumCPU()
 	testCases := []struct {
-		config   componentconfig.KubeProxyConfiguration
+		config   componentconfig.KubeProxyConntrackConfiguration
```

Comment:
Unrelated, but whenever I see tests written like this I want to simplify as follows:

```go
testCases2 := []struct {
  max        int32
  maxPerCore int32
  expected   int
  err        string
}{
  {
    expected: 0,
  },
  ...
}

for i, tc:= range testCases {
  config := componentConfig.KubeProxyConntrackConfiguration{
    Max: tc.max,
    MaxPerCore: tc.maxPerCore,
  }
  ...
}
```

Or is it preferable to define objects in test tables?